pkg/crc/machine/machine.go

```diff
@@ -0,0 +1,177 @@
+package machine
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/errors"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/output"
+
+	"github.com/code-ready/crc/pkg/crc/machine/config"
+	"github.com/code-ready/crc/pkg/crc/machine/libvirt"
+
+	"github.com/code-ready/machine/libmachine"
+	"github.com/code-ready/machine/libmachine/log"
+	"github.com/code-ready/machine/libmachine/state"
+	"github.com/code-ready/machine/libmachine/host"
+)
+
+func init() {
+}
+
+func Start(startConfig StartConfig) (StartResult, error) {
+	result := &StartResult{Name: startConfig.Name}
+
+	// Set libmachine logging
+	setMachineLogging(startConfig.Debug)
+
+	libMachineAPIClient := libmachine.NewClient(constants.MachineBaseDir, constants.MachineCertsDir)
+	defer libMachineAPIClient.Close()
+
+	machineConfig := config.MachineConfig{
+		Name:       startConfig.Name,
+		BundlePath: startConfig.BundlePath,
+		VMDriver:   startConfig.VMDriver,
+		CPUs:	    2,
```

Comment:
We need `4` cpus as default, with `2` there was an issue that one of the container was not able to run because of cpu resource.

Reply:
:-O... my gawd, what minimum requirements are we aiming for?!?